example/contrib/socks-client/socks/impl/async_handshake_v5.hpp

```diff
@@ -36,7 +37,7 @@ template<
     class Buffer,
     class base_type = boost::beast::async_base<
         Handler, typename Stream::executor_type>>
-class socks5_op : public base_type
+class socks5_op : public base_type, public net::coroutine
```

Comment:
`boost::asio::coroutine`, because fauxroutines will certainly never make it into the stdlib :)

Reply:
doesn't this tie the implementation to boost asio? 

Should beast have its own fauxroutine?

Reply:
Well, beast is tied to boost::asio at the moment, so there's no getting around that. I am very confident that Networking-TS / C++ will *never* get fauxroutines so the `net::` prefix is misleading here. This should read `boost::asio::coroutine`. Eventually we might incorporate it into some standalone thing but it can never be in net::.